### ISSUE TITLE
Need help with setup

### ISSUE THREAD

**Issue by prankousky** [koffienl/ESPimaticRF#3]
hi,

i just started setting up a nodeMCU as homeduino device for my pimatic installtion (running on raspberry pi) and can not get it to work. Before, I had used an arduino mega 2560 (clone) with the same receiver and transmitter hardware, which did work. So maybe somebody can help me out here? Any input would be greatly appreciated.

version: ESPimaticRF v0.0.3
config:
mode: homeduino
receiver pin: 0
transmitter pin: 2 [**not** using a transmitter atm while testing, but leaving this empty does not solve my problem]
receive action: send to pimatic
transmit action: both
UDP repeat: 1 [was initially on 1, did not change the value]
wifi credentials according to my wifi setup

on pimatic:
plugin homeduino version: homeduino 0.9.11 (loaded, activated)
driver: serialport
serialDevice: /dev/serial/by-id/usb-1a86_USB2.0-Serial-if00-port0
[did an `ls /dev/serial/by-id` on my raspberry pi, this was the only device found]
baudrate: 115200
enable receiving: yes
receiver pin: 0
transmitter pin: 2
enable DSTSensors: no / not activated
dstSearchAddressPin: default / not activated (12)
connectionTimeout: default / not activated (300000)
debug: yes
apikey: otwirsuhlskfjs47rts (example) [i guess this will only be set in the following nodes because i did not have to enter an apikey in the ESPimaticRF homeduino device]

wires:
node 3V -> 433 receiver VCC
node G -> 433 receiver GND
node D0 -> 433 receiver DATA [there are two DATA pins but afaik either one is fit for use]
node microusb -> raspberry pi usb (device see above)

log in pimatic (debug, info, warn, error)
```
warn [pimatic-homeduino]: Couldn't connect (Serialport not open.), retrying... 2017-05-31 11:00:02
info [pimatic-homeduino]: Connected to homeduino device. 10:50:42
```

when i restart the nodeMCU i get this input:
```
debug [pimatic-homeduino]: data: "Receiving **op** pin 0"
11:10:50 debug [pimatic-homeduino]: data: <myIP>
11:10:50 debug [pimatic-homeduino]: data: "IP address:"
11:10: 50debug [pimatic-homeduino]: data: <myWIFI>
11:10:50 debug [pimatic-homeduino]: data: "Connected to"
11:10:50 debug [pimatic-homeduino]: data: ""
11:10:50 debug [pimatic-homeduino]: data: "."
11:10:49 debug [pimatic-homeduino]: data: "."
11:10:48 debug [pimatic-homeduino]: data: "."
11:10:47 debug [pimatic-homeduino]: data: "."
11:10:46 debug [pimatic-homeduino]: data: "<myWIFI>
11:10:46 debug [pimatic-homeduino]: data: "~ld
�Connecting to :"
```

however, pushing buttons on my remote does nothing. no input is registered (according to *messages* tab in pimatic), device is not triggered.

how is this possible? i have already tried changing receiver pin: 0 to **receiver pin: D0**, but that does not help either.

i am expecting a new receiver and sender module in the mail today, however, those are 5V, not 3.3V, so i am not sure if i will use them on this module (for now, it is set up as a homeduino device, however, once i receive my additional nodes and receivers/transmitters, i will move my pimatic pi to another room and leave this particular nodeMCU there as a node rather than a homeduino device).

i will cross-post this in the homeduino forums and link to this page. thanks in advance for your support =)

**Comment by koffienl** [comment]
Could you test with the latest development version?
https://github.com/koffienl/ESPimaticRF/tree/development

**Comment by prankousky** [comment]
Sorry, my fault. Seems like I did somehow fry the receiver module. It works fine with the new receiver.

Now a completely different problem arose, though.

Can I **only** use 433MHz devices with ESPimaticRF as my homeduino device? I have tried connecting a DHT sensor the same way I would have connected to my arduino, but pimatic did not receive any signals.

Also, would it still be possible to **only** use nodes? I mean not using a nodeMCU as homeduino device, instead using my arduino as the homeduino device, and just have the other nodeMCU, which is setup as a node, talk to pimatic directly via the webinterface?

Or is using a nodeMCU as homeduino device mandatory?

Right now, I have a receiver connected to my node device. I guess "receiver pin 0" was my mistake because the D0 pin is actually considered to be pin 16. Tried either way and did not receive any input in pimatic. In order to test this setup, I disconnected my transmitter and receiver from the homeduino device so that it wouldn't accidentally pick up my remotes signals.

**Comment by prankousky** [comment]
Wow: sorry for being a doofus AGAIN. I already figure out that, indeed, the node can be standalone when it is only supposed to **receive** 433 data. At least disconnecting my *homeduino* node while using with the *node* node worked fine.

**Sending** data, however, seems to require the *homeduino* node to be connected to physical device pimatic is running on, otherwise there is an error (cannot recreate it atm, but something like "cannot access <path of serialdevice>", so regardless of api key etc. it attempts to send from a locally connected device.

so far, so good. so only this question remains for me:
is it still possible to use the other pins of my *homeduino* nodeMCU in order to attach sensors? or maybe even both the *homeduino* one and each *node* one? for example, i need at least three different *node* nodes to work with my *homeduino* node. While I already have to spread those around the house it'd be so amazing if I could just throw in another DHT sensor and get each rooms temperature via wifi signal, as the *node* nodes are transmitting via wifi either way.

Is this possible and/or a planned feature?
+Thanks

**Comment by koffienl** [comment]
No, all the code for DHT and other sensors ar stripped from the original homeduino code. Only for 433 Mhz sending and receiving. At this moment I have no plans for supporting them. Sorry.

You absolutely need a ESP8266 as homeduino connected to pimatic for sending to the nodes.
